src/rascal/structure_managers/adaptor_increase_maxorder.hh

```diff
@@ -205,51 +221,51 @@ namespace rascal {
     }
 
     //! Returns the id of the index-th neighbour atom of a given cluster
-    template <size_t Order, size_t Layer>
-    int get_neighbour_atom_tag(const ClusterRefKey<Order, Layer> & cluster,
+    // tag(felix) this should return an array
+    template <size_t Layer>
+    int get_neighbour_atom_tag(const ClusterRefKey<1, Layer> & cluster,
                                size_t index) const {
-      static_assert(Order < traits::MaxOrder,
-                    "this implementation only handles up to traits::MaxOrder");
-
-      // necessary helper construct for static branching
-      using IncreaseHelper_t =
-          internal::IncreaseHelper<Order == (traits::MaxOrder - 1)>;
-
-      if (Order < (traits::MaxOrder - 1)) {
-        return IncreaseHelper_t::get_neighbour_atom_tag(*this->manager, cluster,
-                                                        index);
-      } else {
-        auto && offset = this->offsets[cluster.get_cluster_index(Layer)];
-        return this->neighbours_atom_tag[offset + index];
-      }
+      return this->manager->get_neighbour_atom_tag(cluster, index);
+    }
+
+    template <size_t Layer>
+    std::array<int, AdditionalOrder - 1>
+    get_neighbour_atom_tag_tt(const ClusterRefKey<1, Layer> & cluster,
```

Comment:
Documentation, specifically how this differ from the above function

Reply:
is it clearer ?

Reply:
yes it can't return just a size_t here.